src/program.cpp

```diff
@@ -507,8 +526,10 @@ std::vector<argument> generic_eval(const module* mod,
                 }));
         }
         assert(results.find(ins) != results.end());
+        // TODO: what order do the arguments to is_compatible_shape() come in?  One
+        // can be dynamic.
         assert(ins->get_shape().any_of_dynamic() or
-               results.at(ins).get_shape() == ins->get_shape());
```

Comment:
Remove the `ins->get_shape().any_of_dynamic()` as the function checks dynamic shapes.